server/routes/companies.js

```diff
@@ -2,14 +2,174 @@
 var express = require('express');
 var router = express.Router();
 
+const db = require('../db')
+
 // Returns test string to verify that Companies server is running.
 router.get('/ping', function(req, res, next) {
   res.send('pong - Companies API');
 });
 
-// GET companies listing.
+//Get all companies hiring a specific major
+//written before /: endpoints since those can have various names
+//Need to check query that sends data for both
+router.get('/filter', function(req, res, next) {
+  const pid = req.query.pid;
+  const mid = req.query.mid;
+
+  if(pid == undefined && mid == undefined){
```

Comment:
I'd rethink some of this logic to consolidate the code and make it more readable. You should be able to take advantage of the AND relationship between the filter parameters.

Reply:
Cleaned up with recent commit